Author: Daniel Baumann <mail@daniel-baumann.ch>
Description: Update debian-cd to syslinux 6.x (except for tools/which_deb).

diff -Naurp debian-cd.orig/debian/control debian-cd/debian/control
--- debian-cd.orig/debian/control
+++ debian-cd/debian/control
@@ -12,7 +12,7 @@ Vcs-Browser: http://anonscm.debian.org/g
 Package: debian-cd
 Architecture: all
 Depends: ${misc:Depends}, curl, perl, dpkg-dev, cpp, libdigest-md5-perl, libdigest-sha-perl, tofrodos, apt (>= 0.3.11.1), make, xorriso | genisoimage, lynx-cur | lynx, grep-dctrl, bc, libcompress-zlib-perl, bzip2, libdpkg-perl
-Recommends: hfsutils, netpbm, syslinux-common, mtools, dosfstools
+Recommends: hfsutils, netpbm, isolinux, syslinux-common, mtools, dosfstools
 Description: Tools for building (Official) Debian CD set
  Debian-cd is the official tool for building Debian CD set since the potato
  release. It was formerly called YACS (for Yet Another CD Script).
diff -Naurp debian-cd.orig/tools/boot/jessie/boot-x86 debian-cd/tools/boot/jessie/boot-x86
--- debian-cd.orig/tools/boot/jessie/boot-x86
+++ debian-cd/tools/boot/jessie/boot-x86
@@ -196,7 +196,7 @@ if [ -n "$ARCHIVE_EXTRACTED_SOURCES" ];
 	find_pkg_file syslinux source >> $CDDIR/../$N.pkgs_extracted
 fi
 
-add_mkisofs_opt $CDDIR/../$N.mkisofs_opts "-isohybrid-mbr syslinux/usr/lib/syslinux/isohdpfx.bin"
+add_mkisofs_opt $CDDIR/../$N.mkisofs_opts "-isohybrid-mbr syslinux/usr/lib/ISOLINUX/isohdpfx.bin"
 
 # In case of a multi-arch CD the script will be called two times. The
 # first time the isolinux dir gets set up for single arch; if it is
@@ -269,13 +269,11 @@ if [ -e boot$N/win32-loader.ini ]; then
 	fi
 fi
 
-cp -f $CDDIR/../syslinux/usr/lib/syslinux/isolinux.bin boot$N/isolinux/
-cp -f $CDDIR/../syslinux/usr/lib/syslinux/vesamenu.c32 boot$N/isolinux/
-if [ -e $CDDIR/../syslinux/usr/lib/syslinux/ldlinux.c32 ]; then
-	cp -f $CDDIR/../syslinux/usr/lib/syslinux/ldlinux.c32 boot$N/isolinux/
-	cp -f $CDDIR/../syslinux/usr/lib/syslinux/libcom32.c32 boot$N/isolinux/
-	cp -f $CDDIR/../syslinux/usr/lib/syslinux/libutil.c32 boot$N/isolinux/
-fi
+cp -f $CDDIR/../syslinux/usr/lib/ISOLINUX/isolinux.bin boot$N/isolinux/
+cp -f $CDDIR/../syslinux/usr/lib/syslinux/modules/bios/vesamenu.c32 boot$N/isolinux/
+cp -f $CDDIR/../syslinux/usr/lib/syslinux/modules/bios/ldlinux.c32 boot$N/isolinux/
+cp -f $CDDIR/../syslinux/usr/lib/syslinux/modules/bios/libcom32.c32 boot$N/isolinux/
+cp -f $CDDIR/../syslinux/usr/lib/syslinux/modules/bios/libutil.c32 boot$N/isolinux/
 
 if [ -z "$MULTIARCH" ]; then
 	# Set up isolinux dir for only the current arch
diff -Naurp debian-cd.orig/tools/boot/jessie/x86-desktop.sh debian-cd/tools/boot/jessie/x86-desktop.sh
--- debian-cd.orig/tools/boot/jessie/x86-desktop.sh
+++ debian-cd/tools/boot/jessie/x86-desktop.sh
@@ -15,7 +15,7 @@ di_syslinux_version() {
 # with vesamenu. Instead, add special default label to automatically
 # select i386/amd64 if user hits enter from help screens.
 multiarch_workaround() {
-	cp -f $CDDIR/../syslinux/usr/lib/syslinux/ifcpu64.c32 boot$N/isolinux/
+	cp -f $CDDIR/../syslinux/usr/lib/syslinux/modules/bios/ifcpu64.c32 boot$N/isolinux/
 	sed -i "/^default install/ s/^/#/" \
 		boot$N/isolinux/txt.cfg || true
 	sed -i "/^default64 amd64-install/ s/^/#/" \
